src/generic/trilinos_eigen_solver.h

```diff
@@ -765,6 +854,116 @@ namespace oomph
       }
     }
 
+    /// Solve the adjoint eigen problem
+    void solve_adjoint_eigenproblem(Problem* const& problem_pt,
+                                    const int& n_eval,
+                                    Vector<std::complex<double>>& eigenvalue,
+                                    Vector<DoubleVector>& eigenvector)
+    {
+      // No access to sigma, so set from sigma real
+      Sigma = Sigma_real;
+
+      // Initially be dumb here
+      Linear_solver_pt = problem_pt->linear_solver_pt();
+
+      // Let's make the initial vector
+      Teuchos::RCP<DoubleMultiVector> initial = Teuchos::rcp(
+        new DoubleMultiVector(1, problem_pt->dof_distribution_pt()));
```

Comment:
Magic number for constructor argument. Is this common for DMV objects, @MatthiasHeilManchester, or should it be changed?

Reply:
What's DMV?  But whatever it is this is really Andrew's stuff so he should comment  @alhazel

Reply:
Oops, I was being lazy, I meant `DoubleMultiVector`!

Reply:
This is just because we are making a single vector i.e. with 1 column, rather than a multi-vector. Put in a more explanatory comment or introduce a variable just for the sake of it?

Reply:
I'd invent a variable so it's clear what the "1" refers to, e.g.

`unsigned nvect=1;`

Reply:
Added,
```
    unsigned multivector_dimension = 1;
```